src/ui/Facet/Facet.ts

```diff
@@ -1528,6 +1528,35 @@ export class Facet extends Component {
     this.dependsOnManager = new DependsOnManager(facetInfo);
   }
 
+  private dependsOnUpdateParentDisplayValue() {
+    if (this.options.dependsOn) {
+      const masterFacetHTML = $$(this.root)
```

Comment:
I think this could be replaced/enhanced using the ComponentTypes utility class:

`ComponentTypes.getAllFacetsInstance(this.root).filter((cmp as Facet)=> cmp.id == this.options.dependsOn)`